File: src/main/java/skywolf46/extrautility/areas/ArrayLocation.java
package skywolf46.extrautility.areas;

import org.bukkit.Location;
import org.bukkit.World;

public class ArrayLocation {
    private double[] xyz = new double[3];

    public ArrayLocation(Location x) {
        xyz = new double[]{
                x.getX(),
                x.getY(),
                x.getZ(),
        };
    }

    public Location toLocation(World wx) {
        return new Location(wx, x(), y(), z());
    }

    public double x() {
        return xyz[0];
    }


    public double y() {
        return xyz[1];
    }

    public double z() {
        return xyz[2];
    }

    public void x(ArrayLocation arx) {
        double cache = x();
        xyz[0] = arx.x();
        arx.xyz[0] = cache;
    }


    public void y(ArrayLocation arx) {
        double cache = y();
        xyz[1] = arx.y();
        arx.xyz[1] = cache;
    }


    public void z(ArrayLocation arx) {
        double cache = z();
        xyz[2] = arx.z();
        arx.xyz[2] = cache;
    }
}

File: minecraft/src/main/kotlin/skywolf46/extrautility/areas/ArrayLocation.kt
package skywolf46.extrautility.areas

import org.bukkit.Location
import org.bukkit.World

class ArrayLocation(val xyz: Array<Double>) : Cloneable {
    constructor(x: Location) : this(arrayOf(x.x, x.y, x.z))

    fun location(wx: World) = Location(wx, xyz[0], xyz[1], xyz[2])

    var x
        get() = xyz[0]
        set(value) {
            xyz[0] = value
        }

    var y
        get() = xyz[1]
        set(value) {
            xyz[1] = value
        }

    var z
        get() = xyz[2]
        set(value) {
            xyz[2] = value
        }

    fun swapX(arx: ArrayLocation) {
        val cache = x
        x = arx.x
        arx.x = cache
    }

    fun swapY(arx: ArrayLocation) {
        val cache = y
        y = arx.y
        arx.y = cache
    }


    fun swapZ(arx: ArrayLocation) {
        val cache = z
        z = arx.z
        arx.z = cache
    }

    public override fun clone(): ArrayLocation {
        return ArrayLocation(xyz)
    }
}